### PR TITLE
feature/jojan_serna/solid

### PR DESCRIPTION
**Pull request by Jojan-Esteban-Serna**
Jojan Esteban Serna Serna

### CODE REVIEW

File: principles/solid/dependencyInjection.md [hunk 1]
```diff
@@ -1,22 +1,58 @@
...
 
+```
+In this example, the UserService class is tightly coupled to the UserRepository class. If the implementation of the UserRepository changes, the UserService class would also need to change.
 ### Good example
 
-[TODO]
+To apply the Dependency Injection principle, you would inject the UserRepository dependency into the UserService class:
+``` java
+public class UserService {
+	private UserRepository userRepository;
```
Comment (zahydo) [inline principles/solid/dependencyInjection.md]: Change to an interface, and explain why it's a  better option using interfaces

**Review by zahydo — CHANGES_REQUESTED** [review]
Fix comments. Good job.

File: principles/solid/dependencyInjection.md (hunk 1 above)
Comment (zahydo) [reply principles/solid/dependencyInjection.md]: The description of all the principles must include the problem, how it solves, and the benefits of applying it.

**Comment by Jojan-Esteban-Serna** [comment]
I think i'm done

**Review by zahydo — APPROVED** [review]
Good Job.

**Pull request merged**